File: app/src/main/java/com/petal/browser/unit/ArchiveBookEmbedManager.java
package com.petal.browser.unit;

import java.net.URLEncoder;

/**
 * ArchiveBookEmbedManager constructs Internet Archive Open Library document & book embed URLs.
 */
public class ArchiveBookEmbedManager {

    /**
     * Constructs Internet Archive Book Embed URL.
     * Endpoint: https://archive.org/embed/
     */
    public static String getBookEmbedUrl(String identifier) {
        if (identifier == null || identifier.trim().isEmpty()) return "";
        try {
            return "https://archive.org/embed/" + URLEncoder.encode(identifier.trim(), "UTF-8");
        } catch (Exception e) {
            return "https://archive.org/embed/" + identifier.trim();
        }
    }
}

File: app/src/main/java/com/petal/browser/unit/ArchiveBookEmbedManager.kt
package com.petal.browser.unit

import java.net.URLEncoder

/**
 * ArchiveBookEmbedManager constructs Internet Archive Open Library document & book embed URLs.
 */
object ArchiveBookEmbedManager {

    @JvmStatic
    fun getBookEmbedUrl(identifier: String?): String {

        if (identifier == null || identifier.trim().isEmpty()) return "";
        return try {
            "https://archive.org/embed/" + URLEncoder.encode(identifier.trim(), "UTF-8");
        
    }
}
